universal-login-react/src/ui/Transfer/Amount/TransferAmount.tsx

```diff
@@ -18,19 +18,22 @@ export interface TransferAmountProps {
 
 export const TransferAmount = ({deployedWallet, onSelectRecipientClick, updateTransferDetailsWith, tokenDetails, transferAmountClassName}: TransferAmountProps) => {
   const [tokenDetailsWithBalance, setTokenDetailsWithBalance] = useState<TokenDetailsWithBalance[]>([]);
-  const [isAmountCorrect, setIsAmountCorrect] = useState(false);
+  const [validAmount, setValidAmount] = useState(true);
+  const [enableButton, setEnableButton] = useState(false);
   const {sdk, contractAddress} = deployedWallet;
 
   useAsyncEffect(() => sdk.subscribeToBalances(contractAddress, setTokenDetailsWithBalance), []);
   const balance = getBalanceOf(tokenDetails.symbol, tokenDetailsWithBalance);
 
   const validateAndUpdateTransferDetails = (amount: string) => {
-    if (balance && amount) {
-      setIsAmountCorrect(Number(amount) <= Number(balance));
+    if (balance && isValidAmount(amount, balance)) {
+      setValidAmount(true);
+      setEnableButton(true);
```

Comment:
Should not be single state? What is difference between validAmount and enableButton?

Reply:
The initial states of above are different. At the beginning you want to disable button but you don't want to show 'Invalid amount' message.